libp2p/connmanager.nim

```diff
@@ -42,10 +42,17 @@ type
   ConnEventHandler* =
     proc(peerId: PeerID, event: ConnEvent): Future[void] {.gcsafe.}
 
-  PeerEvent* {.pure.} = enum
+  PeerEventKind* {.pure.} = enum
     Left,
     Joined
 
+  PeerEvent* = object
+    case kind*: PeerEventKind
+      of PeerEventKind.Joined:
+        initiator*: bool
```

Comment:
so.. same question as usual, why not a `Direction`?

Reply:
I feel that the flag makes more sense here semantically, because it's not about direction, it is who was the initiator of the "session", which is per peer. But it could be `Direction` as well.